staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -350,8 +363,22 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*clientv3.Client, e
 		TLS:                  tlsConfig,
 		Logger:               etcd3ClientLogger,
 	}
-
-	return clientv3.New(cfg)
+	etcdClient, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	consistentListFromCacheEnabled := utilfeature.DefaultFeatureGate.Enabled(genericfeatures.ConsistentListFromCache)
+	if consistentListFromCacheEnabled {
```

Comment:
This doesn't distinguish between "enabled-by-default" and "explicitly enabled by the user", right?

I'm asking as this is not exactly what was defined in the KEP:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2340-Consistent-reads-from-cache#bug-in-etcd-progress-notification

so we should probably ping @deads2k if he's ok with this.

Reply:
Maybe there is another way to check "enabled-by-default" and "explicitly enabled by the user" perhaps checking `GetAll` from `MutableFeatureGate` and get `ConsistentListFromCache` to get Default value and check if it is enabled by user or enabled by default (i'm not sure this works or not). 

What do you think @serathius ?